api/firmware/attestation.go

```diff
@@ -78,23 +79,33 @@ func (device *Device) performAttestation() (bool, error) {
 	if err != nil {
 		return false, err
 	}
+
 	// See parsing below for what the sizes mean.
 	if len(response) < 1+32+64+64+32+64 {
+		device.log.Error(
+			fmt.Sprintf("attestation: response too short. challenge=%x, response=%x", challenge, response), nil)
 		return false, nil
 	}
 	if string(response[:1]) != responseSuccess {
+		device.log.Error(
+			fmt.Sprintf("attestation: expected success. challenge=%x, response=%x", challenge, response), nil)
 		return false, nil
 	}
-	response = response[1:]
+	rsp := response[1:]
 	var bootloaderHash, devicePubkeyBytes, certificate, rootPubkeyIdentifier, challengeSignature []byte
-	bootloaderHash, response = response[:32], response[32:]
-	devicePubkeyBytes, response = response[:64], response[64:]
-	certificate, response = response[:64], response[64:]
-	rootPubkeyIdentifier, response = response[:32], response[32:]
-	challengeSignature = response[:64]
+	bootloaderHash, rsp = rsp[:32], rsp[32:]
+	devicePubkeyBytes, rsp = rsp[:64], rsp[64:]
+	certificate, rsp = rsp[:64], rsp[64:]
+	rootPubkeyIdentifier, rsp = rsp[:32], rsp[32:]
+	challengeSignature = rsp[:64]
 
 	rootPubkeyInfo, ok := attestationPubkeys[hex.EncodeToString(rootPubkeyIdentifier)]
 	if !ok {
+		device.log.Error(fmt.Sprintf(
```

Comment:
That's a funny log.Error method. Consider switching to stdlib logger in the future.